méli-mélo/2024-02-charts/js/charts.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * @title WET-BOEW Charts plugin
+ * @overview Charts plugin based on Charts.js version 4.4.1
+ * @license wet-boew.github.io/wet-boew/License-en.html / wet-boew.github.io/wet-boew/Licence-fr.html
+ * @author garneauma
+ */
+( function( $, window, wb ) {
+"use strict";
+/*
+	* Variable and function definitions.
+	* These are global to the plugin - meaning that they will be initialized once per page,
+	* not once per instance of plugin on the page. So, this is a good place to define
+	* variables that are common to all instances of the plugin on a page.
+	*/
+var componentName = "gc-chart",
+	selector = ".provisional." + componentName,
+	initEvent = "wb-init" + selector,
+	$document = wb.doc,
+	defaults = {},
+
+	/**
+	 * @method init
+	 * @param {jQuery Event} event Event that triggered the function call
+	 */
+	init = function( event ) {
+		// Start initialization
+		// returns DOM object = proceed with init
+		// returns undefined = do not proceed with init (e.g., already initialized)
+		var elm = wb.init( event, componentName, selector ),
+			$elm;
+
+		if ( elm ) {
+			var chart,
+				tableObserver,
+				chartType, chartOpts, chartData,
+				canvasElm, figureElm, figcaptionElm, detailsElm, summaryElm,
+				i18n, i18nText;
+
+			$elm = $( elm );
+
+			// Only initialize the i18nText once
+			if ( !i18nText ) {
+				i18n = wb.i18n;
+				i18nText = {
+					tableMention: i18n( "hyphen" ) + i18n( "tbl-txt" ),
+					tableFollowing: wb.lang == "en" ? " - Chart. Details of the data displayed in this chart can be found in the following table." : " - Graphique. Les détails des données affichées dans ce graphique peuvent être trouvés dans le tableau suivant."
+				};
+			}
+
+			// Creating HTML nodes
+			figureElm = document.createElement( "figure" );
+			figcaptionElm = document.createElement( "figcaption" );
+			figcaptionElm.innerHTML = elm.caption.innerHTML;
+			figcaptionElm.classList.add( "h3" );
+			canvasElm = document.createElement( "canvas" );
+			canvasElm.setAttribute( "aria-label", elm.caption.innerHTML + i18nText.tableFollowing );
+			canvasElm.setAttribute( "role", "img" );
+			detailsElm = document.createElement( "details" );
+			detailsElm.classList.add( "mrgn-tp-md" );
+			summaryElm = document.createElement( "summary" );
+			summaryElm.innerHTML = elm.caption.innerHTML + i18nText.tableMention;
```

Comment:
Comment/idea: May be it could be interesting to leverage the .clone() function of a `<template>` element where it will make easier understand what is the HTML produced by that code.